### PR TITLE
fix: waiting for approval when creating rules

### LINKED ISSUE #245: Approval prompt hangs / does not respond when creating .dexter/RULES.md
## Summary

When starting Dexter in a fresh workspace where `.dexter/` does not yet exist, Dexter attempts to create `.dexter/RULES.md` after the user provides a rule. The terminal UI then gets stuck at `Waiting for approval...`, and pressing `Enter`, `y`, or `yes` does not appear to do anything.

## Environment

- OS: macOS
- Terminal: macOS Terminal
- Command: `bun start`
- Workspace state: fresh checkout / no `.dexter/` directory exists yet
- Terminal size shown in title bar: about `87x28`

## Steps to reproduce

1. Start Dexter in a fresh workspace:

   ```bash
   bun start

### PR DESCRIPTION
**Pull request by swrzalek**
**Fix approval prompt not rendering when creating rules**

### Problem
When the agent requested approval for file-modifying tools (e.g., `write_file` to create `.dexter/RULES.md`), the CLI showed "Waiting for approval..." but never rendered the actual approval overlay with selectable options. Pressing Enter had no effect because the approval prompt was never displayed.

### Root Cause
`renderSelectionOverlay()` was only called from the model selection callback and startup — never from the agent runner's `onChange`. When the agent entered approval state, the working indicator updated but the overlay wasn't triggered.

### Fix
Added `lastPendingApproval` state tracking in the `onChange` callback (`src/cli.ts`). When `pendingApproval` transitions (enters or exits), `renderSelectionOverlay()` is now called to properly show/hide the approval overlay.

### Tests
Added 12 unit tests in `src/controllers/agent-runner.test.ts` covering:
- Approval state transitions (enter/exit)
- `onChange` firing during approval state changes
- `cancelExecution` behavior
- Edge cases (no-op when no approval pending)

### Files Changed
- `src/cli.ts` - Track `pendingApproval` changes and trigger overlay rendering
- `src/controllers/agent-runner.test.ts` - 12 new tests for approval state management

Resolves: #245

### CODE REVIEW

**Comment by swrzalek** [comment]
@virattt ready for review :)

File: src/controllers/agent-runner.test.ts [hunk 1]
```diff
@@ -1,8 +1,61 @@
-import { describe, expect, test } from 'bun:test';
+import { describe, expect, test, mock, beforeEach, afterEach } from 'bun:test';
 import { AgentRunnerController } from './agent-runner.js';
 import { InMemoryChatHistory } from '../utils/in-memory-chat-history.js';
+import type { AgentConfig, AgentEvent, ApprovalDecision } from '../agent/types.js';
+
+/**
+ * Helper to create an AgentRunnerController with a change counter
+ */
+function createController(onChange?: () => void) {
```
Comment (swrzalek) [inline src/controllers/agent-runner.test.ts:9]: we can also create test-utils. I am open.

**Comment by virattt** [comment]
Also fixes: https://github.com/virattt/dexter/issues/266

**Review by virattt — APPROVED** [review]
🚀
